src/main/java/com/hermesworld/ais/galapagos/devcerts/impl/DeveloperCertificateServiceImpl.java

```diff
@@ -119,6 +121,14 @@ public void init(KafkaCluster cluster) {
         return Optional.of(metadata);
     }
 
+    @Override
+    public void clearExpiredDeveloperCertificates(TopicBasedRepository<DevCertificateMetadata> repo,
+            KafkaCluster cluster) {
+        repo.getObjects().stream()
+                .filter(devCert -> devCert.getExpiryDate().isBefore(timeService.getTimestamp().toInstant()))
+                .forEach(expiredCert -> aclUpdater.removeAcls(cluster, Collections.singleton(expiredCert)));
```

Comment:
You are ignoring the CompletableFuture Result here (removeAcls). This whole method must also return a CompletableFuture!